resources/lib/tests/Api/Request/ParametersBuilderTest.php

```diff
@@ -0,0 +1,194 @@
+                        'cat' => 'Category'
+                    ],
+                    'order' => 'price DESC',
+                    'count' => 10
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerSetSearchParams
+     */
+    public function testSetSearchParams($parameters, $category, $expectedParameters)
+    {
+        $requestMock = $this->getMockBuilder(Request::class)->disableOriginalConstructor()->setMethods(null)->getMock();
+
+        $httpRequestMock = $this->getMockBuilder(HttpRequest::class)->disableOriginalConstructor()->setMethods([])->getMock();
+        $httpRequestMock->expects($this->once())->method('all')->willReturn($parameters);
+
+        $categoryMock = null;
+        $parametersBuilderMock = $this->getParametersBuilderMock();
+
+        if ($category) {
+            $categoryMock = $this->getMockBuilder(Category::class)->disableOriginalConstructor()->getMock();
+            $categoryMock->parentCategoryId = $category['parentCategoryId'];
+
+            $details = new \stdClass();
+            $details->name = $category['name'];
+
+            $categoryMock->details = [$details];
+        }
+
+        $result = $parametersBuilderMock->setSearchParams($requestMock, $httpRequestMock, $categoryMock);
+
+        $this->assertEquals($expectedParameters, $result->getParams());
+    }
+
+    public function providerGetCategoryName()
+    {
+        return [
+            [
+                [
+                    2 => ['parentCategoryId' => 1, 'name' => 'Test1'],
+                    1 => ['parentCategoryId' => 0, 'name' => 'Test0']
+                ],
+                'Test1_Test0'
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider providerGetCategoryName
+     */
+    public function testGetCategoryName($categories, $expectedCategoryName)
+    {
+        $mockedCategories = [];
+
+        foreach ($categories as $id => $category) {
+            $categoryMock = $this->getMockBuilder(Category::class)->disableOriginalConstructor()->getMock();
+            $categoryMock->parentCategoryId = $category['parentCategoryId'];
+
+            $details = new \stdClass();
+            $details->name = $category['name'];
+            $categoryMock->details = [$details];
+
+            $mockedCategories[] =[$id, $categoryMock];
+        }
+
+        $this->categoryService->expects($this->any())->method('get')->willReturnMap($mockedCategories);
+
+        $parametersBuilderMock = $this->getParametersBuilderMock();
+
+        $this->assertEquals($expectedCategoryName, $parametersBuilderMock->getCategoryName($mockedCategories[0][1]));
+    }
+
+    /**
+     * @param array|null $methods
+     * @return ParametersBuilder|MockObject
+     */
+    protected function getParametersBuilderMock($methods = null)
+    {
+        if (!is_array($methods)) {
+            $methods = [];
+        }
+
+        $methods = array_merge($methods, ['getCategoryService']);
```

Comment:
I think it is safe to use `$methosds[] = 'getCategoryService'` instead.

Reply:
I will leave this like it is for the future tests.

Reply:
I had to read this 5 times before I understood it :smile: 
We can leave this as it is, since your point is valid.